node-client/src/config.ts

```diff
@@ -318,22 +320,25 @@ export class Config {
     public static SERVICEACCOUNT_TOKEN_PATH =
     Config.SERVICEACCOUNT_ROOT + '/token';
 
-    public static fromFile(filename: string): api.Core_v1Api {
+    public static apiFromFile<T extends ApiType>(filename: string, apiClientType: ApiConstructor<T>): T {
```

Comment:
Let's keep the original functions too so that we don't break users with existing code who upgrade the library.